glide-core/src/client/value_conversion.rs

```diff
@@ -640,6 +685,254 @@ mod tests {
+                        Value::BulkString(b"streamid-3".to_vec()),
+                        Value::Array(vec![
+                            Value::BulkString(b"field3".to_vec()),
+                            Value::BulkString(b"value3".to_vec()),
+                        ]),
+                    ]),
+                ]),
+            ]),
+        ];
+
+        // convert to a map value like this:
+        // #1) "key1"
+        //    #1) "streamid-1"
+        //         1) "f1"
+        //         2) "v1"
+        //         3) "f2"
+        //         4) "v2"
+        //    #2) "streamid-2"
+        //    ...
+        // #2) "key2"
+        // ...
+        let converted_map = convert_to_expected_type(
+            Value::Array(array_of_arrays),
+            Some(ExpectedReturnType::Map {
+                key_type: &Some(ExpectedReturnType::BulkString),
+                value_type: &Some(ExpectedReturnType::Map {
+                    key_type: &Some(ExpectedReturnType::BulkString),
+                    value_type: &Some(ExpectedReturnType::ArrayOfStrings),
+                }),
+            }),
+        )
+        .unwrap();
+
+        let converted_map = if let Value::Map(map) = converted_map {
+            map
+        } else {
+            panic!("Expected a Map, but got {:?}", converted_map);
+        };
+        // expect 2 keys
+        assert_eq!(converted_map.len(), 2);
+
+        let (key, value) = &converted_map[0];
+        assert_eq!(Value::BulkString(b"key1".to_vec()), *key);
+        assert_eq!(
+            Value::Map(vec![(
+                Value::BulkString(b"streamid-1".to_vec()),
+                Value::Array(vec![
+                    Value::BulkString(b"field1".to_vec()),
+                    Value::BulkString(b"value1".to_vec()),
+                ]),
+            ),]),
+            *value,
+        );
+
+        let (key, value) = &converted_map[1];
+        assert_eq!(*key, Value::BulkString(b"key2".to_vec()));
+        assert_eq!(
+            Value::Map(vec![
+                (
+                    Value::BulkString(b"streamid-2".to_vec()),
+                    Value::Array(vec![
+                        Value::BulkString(b"field21".to_vec()),
+                        Value::BulkString(b"value21".to_vec()),
+                        Value::BulkString(b"field22".to_vec()),
+                        Value::BulkString(b"value22".to_vec()),
+                    ]),
+                ),
+                (
+                    Value::BulkString(b"streamid-3".to_vec()),
+                    Value::Array(vec![
+                        Value::BulkString(b"field3".to_vec()),
+                        Value::BulkString(b"value3".to_vec()),
+                    ]),
+                ),
+            ]),
+            *value,
+        );
+    }
+
+    #[test]
+    fn test_convert_map_with_inner_array_to_map_of_maps_using_ExpectedReturnType_Map() {
+        // in RESP3, we get a map of arrays value like this:
```

Comment:
Ideally to test how RESP2 response converted and how RESP3 too

Reply:
I split the tests into two... is that okay?